### PR TITLE
Change the service registration

### PR DESCRIPTION
**Pull request by Kamidu**

### CODE REVIEW

**Comment by Kamidu** [comment]
Enable Oauth authentication with java 8 support

File: components/identity-extensions/backend-oauth-authenticator/src/main/java/org/wso2/carbon/identity/authenticator/backend/oauth/internal/OauthAuthenticatorServiceComponent.java [hunk 1]
```diff
@@ -20,25 +20,27 @@
...
 
-/**
- * @scr.component component.name="org.wso2.carbon.identity.authenticator.backend.oauth.OauthAuthenticator" immediate="true"
- */
-@SuppressWarnings("unused")
-public class OauthAuthenticatorServiceComponent {
 
+public class OauthAuthenticatorServiceComponent  implements BundleActivator {
+
+    private ServiceRegistration pipServiceRegRef;
```
Comment (harshanL) [inline components/identity-extensions/backend-oauth-authenticator/src/main/java/org/wso2/carbon/identity/authenticator/backend/oauth/internal/OauthAuthenticatorServiceComponent.java:33]: What is the purpose of having "pip" in this variable name? I think its unnecessary.

Comment (Kamidu) [inline components/identity-extensions/backend-oauth-authenticator/src/main/java/org/wso2/carbon/identity/authenticator/backend/oauth/internal/OauthAuthenticatorServiceComponent.java:33]: These formatting issues has been fixed in a recent PR.In a latest PR this class has been replaced.the master branch now contains a new version of OAuthAuthenticatorServiceComponent.java

File: components/identity-extensions/backend-oauth-authenticator/src/main/java/org/wso2/carbon/identity/authenticator/backend/oauth/internal/OauthAuthenticatorServiceComponent.java [hunk 2]
```diff
@@ -20,25 +20,27 @@
...
     private static final Log log = LogFactory.getLog(OauthAuthenticatorServiceComponent
             .class);
 
-    protected void activate(ComponentContext ctxt) {
+    @Override
+    public void start(BundleContext bundleContext) throws Exception {
+        log.info("Initiating");
```
Comment (prabathabey) [inline components/identity-extensions/backend-oauth-authenticator/src/main/java/org/wso2/carbon/identity/authenticator/backend/oauth/internal/OauthAuthenticatorServiceComponent.java:39]: Add a more meaningful info message.

Comment (Kamidu) [inline components/identity-extensions/backend-oauth-authenticator/src/main/java/org/wso2/carbon/identity/authenticator/backend/oauth/internal/OauthAuthenticatorServiceComponent.java:39]: This was change to a log message and update to "Starting Backend OAuthAuthenticator Framework Bundle"  in a recent PR

File: components/webapp-authenticator-framework/org.wso2.carbon.webapp.authenticator.framework/src/main/java/org/wso2/carbon/webapp/authenticator/framework/authenticator/OAuthAuthenticator.java [hunk 3]
```diff
@@ -81,8 +80,8 @@ public AuthenticationInfo authenticate(Request request, Response response) {
             authenticationInfo.setStatus(Status.CONTINUE);
         }
         String apiVersion = tokenizer.nextToken();
-        String authLevel = authenticator.getResourceAuthenticationScheme(context, apiVersion, requestUri, requestMethod);
-        //String authLevel = "any";
+        //String authLevel = authenticator.getResourceAuthenticationScheme(context, apiVersion, requestUri, requestMethod);
```
Comment (sumuditha-viraj) [inline components/webapp-authenticator-framework/org.wso2.carbon.webapp.authenticator.framework/src/main/java/org/wso2/carbon/webapp/authenticator/framework/authenticator/OAuthAuthenticator.java:83]: Remove commented codes

Comment (Kamidu) [inline components/webapp-authenticator-framework/org.wso2.carbon.webapp.authenticator.framework/src/main/java/org/wso2/carbon/webapp/authenticator/framework/authenticator/OAuthAuthenticator.java:83]: This commented lines were removed and refactored in a recent PR

**Pull request merged**